python/cudf/cudf/core/udf/utils.py

```diff
@@ -87,35 +89,35 @@ def _get_udf_return_type(argty, func: Callable, args=()):
     return result
 
 
-def _is_jit_supported_type(dtype):
+def _is_jit_supported_type(dtype, supported_types):
     # category dtype isn't hashable
     if isinstance(dtype, CategoricalDtype):
         return False
-    return str(dtype) in JIT_SUPPORTED_TYPES
+    return str(dtype) in supported_types
```

Comment:
Why is this change needed, it seems you just push the requirement to provide `JIT_SUPPORTED_TYPES` as an argument to all callers?

Reply:
Not sure if this was the best solution but I wanted a way of having these functions work for different UDF APIs without having to have separate copies for each API that may or may not support the same set of input columns. For instance we have strings in `DataFrame.apply` but we'll not have it in `Groupby.apply`.

Reply:
I think at the very least we could rename this utility to reflect its new responsibility, e.g., something like: `_is_jit_supported_type(dtype, supported_types)` -> `_is_dtype_equal(dtype, tuple_of_dtypes)`

Reply:
This has been inlined and removed.